dependencies/testing/testing.go

```diff
@@ -0,0 +1,98 @@
+package testing
```

Comment:
Can you add some simple unit tests for the methods in this package? Something along the lines of create a context, call the various methods call Check etc.

Reply:
Done.